internal/core/metadata/rest_deviceservice_test.go

```diff
@@ -560,6 +576,87 @@ func TestUpdateAdminStateByName(t *testing.T) {
 	}
 }
 
+func TestUpdateServiceLastReportedById(t *testing.T) {
+
+	tests := []struct {
+		name           string
+		request        *http.Request
+		dbMock         interfaces.DBClient
+		expectedStatus int
+	}{
+		{
+			"ok",
+			createDeviceServiceRequestWithBody(
+				http.MethodPut, testDeviceServiceLastUpdated, map[string]string{ID: testDeviceServiceLastUpdatedId, LASTREPORTED: strconv.Itoa(lastReported)}),
+			createMockWithOutlines([]mockOutline{
+				{"GetDeviceServiceById", testDeviceServiceLastUpdatedId, testDeviceServiceLastUpdated, nil},
+				{"UpdateDeviceService", testDeviceServiceLastUpdated, nil, nil},
+			}),
+			http.StatusOK,
+		},
+		{
+			"request validation error",
+			createDeviceServiceRequestWithBody(http.MethodPut, testDeviceServiceLastUpdated, map[string]string{ID: testDeviceServiceLastUpdatedId, LASTREPORTED: ""}),
+			createMockWithOutlines([]mockOutline{
+				{"GetDeviceServiceById", testDeviceServiceLastUpdatedId, testDeviceServiceLastUpdated, testError},
+				{"UpdateDeviceService", testDeviceServiceLastUpdated, nil, testError},
+			}),
+			http.StatusBadRequest,
+		},
+		{
+			"device service lookup error",
+			createDeviceServiceRequestWithBody(http.MethodPut, testDeviceServiceLastUpdatedNotExist, map[string]string{ID: testDeviceServiceLastUpdatedNotExistId, LASTREPORTED: strconv.Itoa(lastReported)}),
+			createMockWithOutlines([]mockOutline{
+				{"GetDeviceServiceById", testDeviceServiceLastUpdatedNotExistId, contract.DeviceService{}, db.ErrNotFound},
+			}),
+			http.StatusInternalServerError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rr := httptest.NewRecorder()
+			var loggerMock = logger.NewMockClient()
```

Comment:
For consistency, can we use the `:=` operator here. Our code base tends to favor `:=` over `var`. This also applies to other uses of `var` in these changes.